apps/synapse-oauth-signin/src/OAuth2Form.tsx

```diff
@@ -340,16 +296,17 @@ export function OAuth2Form() {
           description="This app has not been verified by Sage Bionetworks yet."
           primaryButtonConfig={{
             text: 'Back to Safety',
-            onClick: onGoBack,
+            onClick: () => {
+              // The client verification warning appears before the user has a chance to sign in
+              // So there is no risk of going 'back' to an external IdP to sign in to synapse
+              window.history.back()
+            },
```

Comment:
This is the only place `onGoBack` was called. This behavior relies on the assumption that the browser's last location was the client application, which is not always true. To prevent the risk of future error, I decided to remove the function definition and inline this behavior, with a comment.